File: app/src/main/java/xyz/lrhm/komakdast/core/util/legacy/LengthManager.java
package xyz.lrhm.komakdast.core.util.legacy;

import android.content.Context;
import android.content.res.Resources;
import android.graphics.BitmapFactory;
import android.util.Pair;

import xyz.lrhm.komakdast.R;

public class LengthManager {
    //    private static boolean initialized = false;
    private int screenHeight;
    private int screenWidth;
    private Context context;
//    private Tools tools;

    public LengthManager(Context context) {
        this.context = context;
        screenHeight = getScreenHeight(context);
        screenWidth = getScreenWidth(context);
//        tools = new Tools(context);
    }

    private int getScreenWidth(Context context) {

        return context.getResources().getDisplayMetrics().widthPixels;
    }

    private int getScreenHeight(Context context) {
        return context.getResources().getDisplayMetrics().heightPixels;

    }

    public int getScreenWidth() {
        return screenWidth;
    }

    public int getScreenHeight() {
        return screenHeight;
    }

    public Pair<Integer, Integer> getResourceDimensions(int resourceId) {
        Resources resources = context.getResources();
        BitmapFactory.Options bounds = new BitmapFactory.Options();
        bounds.inJustDecodeBounds = true;
        BitmapFactory.decodeResource(resources, resourceId, bounds);
        return new Pair<>(bounds.outWidth, bounds.outHeight);
    }

    public int getHeightWithFixedWidth(int resourceId, int width) {
        final Pair<Integer, Integer> dimensions = getResourceDimensions(resourceId);
        return dimensions.second * width / dimensions.first;
    }

    public int getWidthWithFixedHeight(int resourceId, int height) {
        final Pair<Integer, Integer> dimensions = getResourceDimensions(resourceId);
        return dimensions.first * height / dimensions.second;
    }

    public int getHeaderHeight() {
        return (int) (getScreenHeight() * 0.12f);
    }

//    public int getTabsHeight() {
//
//        return getHeightWithFixedWidth(R.drawable.tabbar_background, getScreenWidth());
//    }
//
//    public int getTabBarShadeHeight() {
//        return getHeightWithFixedWidth(R.drawable.shadow_top, getScreenWidth());
//    }

//    public int getTabBarHeight() {
//        return getTabsHeight() + getTabBarShadeHeight();
//    }

    public int getPageLevelCount() {
        return getPageRowCount() * getPageColumnCount();
    }

    public int getPageColumnCount() {
        return 4;
    }

    public int getPageRowCount() {
        return 4;
    }

    public int getAlphabetButtonSize() {
        return getScreenWidth() / 8;
    }

    public float getAlphabetFontSize() {
        return getScreenWidth() / 14;
    }

    public int getSolutionButtonSize() {
        return getScreenWidth() / 12;
    }

    public float getSolutionFontSize() {
        return getScreenWidth() / 24;
    }

    public int getFragmentHeight() {
        return getScreenHeight() - getHeaderHeight();
    }

    public int getLevelsBackTopHeight() {
        return getHeightWithFixedWidth(R.drawable.top_seperator, getScreenWidth());
    }

    public int getLevelsBackBottomHeight() {
        return getHeightWithFixedWidth(R.drawable.bottom_seperator, getScreenWidth());
    }

    public int getLevelsViewpagerHeight() {
        return getPageRowCount() * getLevelFrameHeight() + 2 * getLevelsGridViewTopAndBottomPadding();
    }

    public int getLevelsGridViewLeftRightPadding() {
        return getScreenWidth() / 20;
    }

    public int getLevelFrameWidth() {
        return (getScreenWidth() - 2 * getLevelsGridViewLeftRightPadding()) / 4;
    }

    public int getLevelFrameHeight() {
        return (int) (getScreenWidth() * 0.235);
        //getHeightWithFixedWidth(R.drawable.level_unlocked, getLevelFrameWidth());
    }

    public int getLevelImageFrameWidth() {
        return getScreenWidth() * 91 / 100;
    }

    public int getLevelImageFrameHeight() {
        return (int) (getLevelImageFrameWidth() * (3.05 / 4.f));
    }


    public int getLevelThumbnailPadding() {
        return getLevelFrameWidth() * 10 / 100;
    }

    public int getLevelsGridViewTopAndBottomPadding() {
        return 0;
    }

    public int getLevelImageWidth() {
        return getLevelImageFrameWidth() * 927 / 997;
    }

    public int getLevelImageHeight() {
        return getLevelImageFrameHeight() * 642 / 704;
    }

    public int getIndicatorBigSize() {
        return getScreenWidth() / 15;
    }

    public int getIndicatorSmallSize() {
        return getScreenWidth() / 30;
    }

    public int getStoreDialogMargin() {
        return getScreenWidth() / 20;
    }

    public int getStoreDialogPadding() {
        return getScreenWidth() / 15;
    }

    public int getStoreItemsInnerWidth() {
        return getScreenWidth() - 2 * (getStoreDialogMargin() + getStoreDialogPadding());
    }

    public float getStoreItemFontSize() {
        return getScreenWidth() * 60 / 1000;
    }


    public int getStoreItemWidth() {
        return getScreenWidth() * 70 / 100;
    }


    public int getCheatButtonWidth() {
        return getLevelImageWidth();
    }

    public float getCheatButtonFontSize() {
        return getScreenWidth() / 15;
    }

    public int getShopTitleWidth() {
        return getScreenWidth() / 2;
    }

    public int getToastWidth() {
        return getScreenWidth() * 90 / 100;
    }

    public float getToastFontSize() {
        return getScreenWidth() / 15f;
    }

    public int getCheatButtonSize() {
        return getScreenWidth() / 7;
    }

    public int getVideoPlayButtonSize() {
        return getLevelImageWidth() / 2;
    }

    public int getShopTitleBottomMargin() {
        return getShopTitleWidth() / 8;
    }

    public int getTickViewSize() {
        return getScreenWidth() / 3;
    }

    public int getLevelFinishedButtonsSize() {
        return getScreenWidth() / 6;
    }

    public float getLevelSolutionTextSize() {
        return getScreenWidth() / 15f;
    }


    public float getLevelAuthorTextSize() {
        return getScreenWidth() / 18f;
    }

    public int getLevelFinishedDialogPadding() {
        return getScreenWidth() / 10;
    }

    public int getOneFingerDragWidth() {
        return getLevelImageFrameWidth() / 4;
    }

    public int getResourceLockWidth() {
        return getLevelImageFrameWidth() / 3;
    }

    public int getToastPadding() {
        return getToastWidth() / 17;
    }

    public float getPackagePurchaseTitleSize() {
        return getScreenWidth() / 13f;
    }

    public float getPackagePurchaseDescriptionSize() {
        return getScreenWidth() / 19f;
    }

    public float getPackagePurchaseItemTextSize() {
        return getScreenWidth() / 22f;
    }

    public int getPackagePurchaseItemWidth() {
        return getScreenWidth() * 60 / 100;
    }

    public int getPackagePurchaseItemHeight() {
        return getPackagePurchaseItemWidth() * 504 / 1809;
    }

    public int getLoadingEhemWidth() {
        return getScreenWidth() / 3;
    }

    public int getLoadingToiletWidth() {
        return getScreenWidth() / 11;
    }

    public int getLoadingEhemPadding() {
        return getScreenWidth() / 11;
    }

//    public int getPackagesListColumnCount() {
//        return Math.min(Math.max((int) (tools.convertPixelsToDp(getScreenWidth()) / 150), 1), 3);
//    }

    public int getLevelFinishedDialogTopPadding() {
        return getScreenWidth() / 30;
    }

    public int getPrizeBoxSize() {
        return getScreenWidth() / 5;
    }

    public float getTipTextSize() {
        return getScreenWidth() / 12.F;
    }

    public int getTipWidth() {
        return getScreenWidth() * 8 / 10;
    }

    public float getPlayStopButtonFontSize() {
        return getScreenWidth() / 3;
    }

    public int getAdViewPagerHeight() {
        return getScreenWidth() * 579 / 1248;
    }

//    public int getPackageIconSize() {
//        return getScreenWidth() / getPackagesListColumnCount();
//    }

    public float getPlaceHolderTextSize() {
        return getScreenWidth() / 14.F;
    }
}

File: app/src/main/java/xyz/lrhm/komakdast/core/util/legacy/LengthManager.kt
package xyz.lrhm.komakdast.core.util.legacy

import android.content.Context
import android.graphics.BitmapFactory
import android.util.Pair
import dagger.hilt.android.qualifiers.ApplicationContext
import xyz.lrhm.komakdast.R
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class LengthManager @Inject constructor(@ApplicationContext val context: Context) {
    //    private static boolean initialized = false;

    val screenHeight = context.resources.displayMetrics.heightPixels


    val screenWidth = context.resources.displayMetrics.widthPixels


    fun getResourceDimensions(resourceId: Int): Pair<Int, Int> {
        val resources = context.resources
        val bounds = BitmapFactory.Options()
        bounds.inJustDecodeBounds = true
        BitmapFactory.decodeResource(resources, resourceId, bounds)
        return Pair(bounds.outWidth, bounds.outHeight)
    }

    fun getHeightWithFixedWidth(resourceId: Int, width: Int): Int {
        val dimensions = getResourceDimensions(resourceId)
        return dimensions.second * width / dimensions.first
    }

    fun getWidthWithFixedHeight(resourceId: Int, height: Int): Int {
        val dimensions = getResourceDimensions(resourceId)
        return dimensions.first * height / dimensions.second
    }

    val headerHeight: Int
        get() = (screenHeight * 0.12f).toInt()

    //    public int getTabsHeight() {
//
//        return getHeightWithFixedWidth(R.drawable.tabbar_background, getScreenWidth());
//    }
//
//    public int getTabBarShadeHeight() {
//        return getHeightWithFixedWidth(R.drawable.shadow_top, getScreenWidth());
//    }
//    public int getTabBarHeight() {
//        return getTabsHeight() + getTabBarShadeHeight();
//    }
    val pageLevelCount: Int
        get() = getPageRowCount() * getPageColumnCount()

    fun getPageColumnCount(): Int {
        return 4
    }

    fun getPageRowCount(): Int {
        return 4
    }

    fun getAlphabetButtonSize(): Int {
        return screenWidth / 8
    }

    fun getAlphabetFontSize(): Float {
        return (screenWidth / 14).toFloat()
    }

    fun getSolutionButtonSize(): Int {
        return screenWidth / 12
    }

    fun getSolutionFontSize(): Float {
        return (screenWidth / 24).toFloat()
    }

    fun getFragmentHeight(): Int {
        return screenHeight - headerHeight
    }

    fun getLevelsBackTopHeight(): Int {
        return getHeightWithFixedWidth(R.drawable.top_seperator, screenWidth)
    }

    fun getLevelsBackBottomHeight(): Int {
        return getHeightWithFixedWidth(R.drawable.bottom_seperator, screenWidth)
    }

    fun getLevelsViewpagerHeight(): Int {
        return getPageRowCount() * getLevelFrameHeight() + 2 * getLevelsGridViewTopAndBottomPadding()
    }

    fun getLevelsGridViewLeftRightPadding(): Int {
        return screenWidth / 20
    }

    fun getLevelFrameWidth(): Int {
        return (screenWidth - 2 * getLevelsGridViewLeftRightPadding()) / 4
    }

    fun getLevelFrameHeight(): Int {
        return (screenWidth * 0.235).toInt()
        //getHeightWithFixedWidth(R.drawable.level_unlocked, getLevelFrameWidth());
    }

    fun getLevelImageFrameWidth(): Int {
        return screenWidth * 91 / 100
    }

    fun getLevelImageFrameHeight(): Int {
        return (getLevelImageFrameWidth() * (3.05 / 4f)).toInt()
    }

    fun getLevelThumbnailPadding(): Int {
        return getLevelFrameWidth() * 10 / 100
    }

    fun getLevelsGridViewTopAndBottomPadding(): Int {
        return 0
    }

    fun getLevelImageWidth(): Int {
        return getLevelImageFrameWidth() * 927 / 997
    }

    fun getLevelImageHeight(): Int {
        return getLevelImageFrameHeight() * 642 / 704
    }

    fun getIndicatorBigSize(): Int {
        return screenWidth / 15
    }

    fun getIndicatorSmallSize(): Int {
        return screenWidth / 30
    }

    fun getStoreDialogMargin(): Int {
        return screenWidth / 20
    }

    fun getStoreDialogPadding(): Int {
        return screenWidth / 15
    }

    fun getStoreItemsInnerWidth(): Int {
        return screenWidth - 2 * (getStoreDialogMargin() + getStoreDialogPadding())
    }

    fun getStoreItemFontSize(): Float {
        return (screenWidth * 60 / 1000).toFloat()
    }

    fun getStoreItemWidth(): Int {
        return screenWidth * 70 / 100
    }

    fun getCheatButtonWidth(): Int {
        return getLevelImageWidth()
    }

    fun getCheatButtonHeight(): Int {
        return getHeightWithFixedWidth(R.drawable.cheat_right, getCheatButtonWidth())
    }

    fun getCheatButtonFontSize(): Float {
        return (screenWidth / 15).toFloat()
    }

    fun getShopTitleWidth(): Int {
        return screenWidth / 2
    }

    fun getToastWidth(): Int {
        return screenWidth * 90 / 100
    }

    fun getToastFontSize(): Float {
        return screenWidth / 15f
    }

    fun getCheatButtonSize(): Int {
        return screenWidth / 7
    }

    fun getVideoPlayButtonSize(): Int {
        return getLevelImageWidth() / 2
    }

    fun getShopTitleBottomMargin(): Int {
        return getShopTitleWidth() / 8
    }

    fun getTickViewSize(): Int {
        return screenWidth / 3
    }

    fun getLevelFinishedButtonsSize(): Int {
        return screenWidth / 6
    }

    fun getLevelSolutionTextSize(): Float {
        return screenWidth / 15f
    }

    fun getLevelAuthorTextSize(): Float {
        return screenWidth / 18f
    }

    fun getLevelFinishedDialogPadding(): Int {
        return screenWidth / 10
    }

    fun getOneFingerDragWidth(): Int {
        return getLevelImageFrameWidth() / 4
    }

    fun getResourceLockWidth(): Int {
        return getLevelImageFrameWidth() / 3
    }

    fun getToastPadding(): Int {
        return getToastWidth() / 17
    }

    fun getPackagePurchaseTitleSize(): Float {
        return screenWidth / 13f
    }

    fun getPackagePurchaseDescriptionSize(): Float {
        return screenWidth / 19f
    }

    fun getPackagePurchaseItemTextSize(): Float {
        return screenWidth / 22f
    }

    fun getPackagePurchaseItemWidth(): Int {
        return screenWidth * 60 / 100
    }

    fun getPackagePurchaseItemHeight(): Int {
        return getPackagePurchaseItemWidth() * 504 / 1809
    }

    fun getLoadingEhemWidth(): Int {
        return screenWidth / 3
    }

    fun getLoadingToiletWidth(): Int {
        return screenWidth / 11
    }

    fun getLoadingEhemPadding(): Int {
        return screenWidth / 11
    }

    //    public int getPackagesListColumnCount() {
//        return Math.min(Math.max((int) (tools.convertPixelsToDp(getScreenWidth()) / 150), 1), 3);
//    }
    fun getLevelFinishedDialogTopPadding(): Int {
        return screenWidth / 30
    }

    fun getPrizeBoxSize(): Int {
        return screenWidth / 5
    }

    fun getTipTextSize(): Float {
        return screenWidth / 12f
    }

    fun getTipWidth(): Int {
        return screenWidth * 8 / 10
    }

    fun getPlayStopButtonFontSize(): Float {
        return (screenWidth / 3).toFloat()
    }

    fun getAdViewPagerHeight(): Int {
        return screenWidth * 579 / 1248
    }

    //    public int getPackageIconSize() {
//        return getScreenWidth() / getPackagesListColumnCount();
//    }
    fun getPlaceHolderTextSize(): Float {
        return screenWidth / 14f
    }
    
}